webui/react/src/pages/WorkspaceDetails/WorkspaceQuickSearch.tsx

```diff
@@ -0,0 +1,150 @@
+import { Input, Modal, Tree } from 'antd';
+import type { DefaultOptionType } from 'rc-tree-select/lib/TreeSelect';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+
+import Link from 'components/Link';
+import { paths } from 'routes/utils';
+import { getWorkspaceProjects, getWorkspaces } from 'services/api';
+import Icon from 'shared/components/Icon/Icon';
+import Message, { MessageType } from 'shared/components/Message';
+import Spinner from 'shared/components/Spinner';
+import { ErrorLevel, ErrorType } from 'shared/utils/error';
+import { routeToReactUrl } from 'shared/utils/routes';
+import { Project, Workspace } from 'types';
+import handleError from 'utils/error';
+
+import css from './WorkspaceQuickSearch.module.scss';
+
+interface Props {
+  children: React.ReactChild;
+}
+
+const WorkspaceQuickSearch: React.FC<Props> = ({ children }: Props) => {
+  const [ searchText, setSearchText ] = useState<string>('');
+  const [ workspaceMap, setWorkspaceMap ] = useState<Map<Workspace, Project[]>>(new Map());
+  const [ isLoading, setIsLoading ] = useState(true);
+  const [ isModalVisible, setIsModalVisible ] = useState(false);
+  const canceler = useRef(new AbortController());
+
+  const fetchData = useCallback(async () => {
+    try {
+      const workspaceResponse = await getWorkspaces(
+        { limit: 0, sortBy: 'SORT_BY_NAME' },
+        { signal: canceler.current.signal },
+      );
+      const filteredWorkspaces = workspaceResponse.workspaces.filter((w) => !w.immutable);
+      const projectAPIList = filteredWorkspaces
+        .map((workspace) =>
+          getWorkspaceProjects(
+            { id: workspace.id, sortBy: 'SORT_BY_NAME' },
+            { signal: canceler.current.signal },
+          ));
+      const projectResponse = (await Promise.all(projectAPIList))
```

Comment:
This is a concern. If user has lots of projects, we need to call the api a lot. Wanna make it 1 api call

Reply:
yeah ideally we could have one API call, we would need to make sure it respects RBAC though. 

@NicholasBlaskey if we wanted to make an API to get all projects in all workspaces for which a user is authorized, how might be go about doing that?

Reply:
Thanks for the info. I'll make the change in different PR to make this PR simple